### ISSUE TITLE
games with Valve's CEG drm fails to launch, patch needed

### ISSUE THREAD

**Issue by ahjolinna** [kakra/wine-proton#11]
I just wanted to try [warhammer 40k space marine](https://store.steampowered.com/app/55150/Warhammer_40000_Space_Marine/) (55150) by Relic/SEGA, but apparently it wont run because of some bs DRM, but apparently it has been patched in staging.
here's the relevant bugzilla:  https://bugs.winehq.org/show_bug.cgi?id=32515

as I know you are busy, I just wanted to add this to your "to-do" list. as long as I remember this one....I do have other things I need to "report/talk" about, like for example 32-bit prefix support so I can play the whole Dawn Of War1 series (and not just the 2 last standalone expansions)

**Comment by kakra** [comment]
32 bit games should work... But it hasn't been tested a lot. DXVK could be problematic here because of memory usage. If there's anything specific, feel free to report an issue.

I'll look into the DRM thing soon...

**Comment by ahjolinna** [comment]
about the 32bit thing, at least with games like Dawn of War1 ([GOTY](https://www.protondb.com/app/4570), or/and main game, + [winter assault ](https://www.protondb.com/app/9310)expansion) don't work, but  for some reason the newer expansions do work just fine aka [Dark Crusade](https://www.protondb.com/app/4580) and [Soulstorm](https://www.protondb.com/app/9450)

here is the proton [32bit prefix support feature request](https://github.com/ValveSoftware/Proton/issues/449)

**Comment by kakra** [comment]
@ahjolinna Ah okay that problem. The way it works now is: The proton launcher script looks at the exe file to be started: If it is 32bit, it initializes a 32bit prefix, if it is 64bit, it initializes a 64bit prefix. If it is anything else (e.g. a batch file) or cannot be detected (e.g., an uncommon exe format), it defaults to 64bit.

I think that's the problem here. The prefix may also have been created with an earlier version which didn't do this detection. In that case you'll have to kill the prefix first.

I could put a manual override into my own launcher easily as it is currently planned.

PS: I've added the idea here: https://github.com/kakra/wine-proton/projects/3

**Comment by Saroufim** [comment]
I'm not sure if this is the same but the Elder Scrolls Online does not launch either. Bethesda's launcher works but ESO64.exe crashes immediately on the spot while it works with the regular version of proton.

**Comment by kakra** [comment]
Is there any game available in Steam as demo or free version which shows this problem? And can be reproduced with such a version?

**Comment by ahjolinna** [comment]
I doubt any free/demo game has a DRM, as that wouldn't make no sense ...but I can check. I will also look if there is any game that would be REALLY cheap to buy

**Comment by ahjolinna** [comment]
well after a quick look; at least [Spec Ops: The Line](https://store.steampowered.com/app/50300/Spec_Ops_The_Line/), [Mafia 2](https://store.steampowered.com/app/50130/Mafia_II/), [XCOM: Enemy Unknown](https://store.steampowered.com/app/200510/XCOM_Enemy_Unknown/) has a demos

other games that has it this drm (according to the bug report I posted);

- [Bureau: XCOM Declassified](https://store.steampowered.com/app/65930/The_Bureau_XCOM_Declassified/)
- [Red Faction®: Armageddon™](https://store.steampowered.com/app/55110/Red_Faction_Armageddon/)
- [Deus Ex: Human Revolution](https://store.steampowered.com/app/238010/Deus_Ex_Human_Revolution__Directors_Cut/)
- [Lord of the Rings: War in the North](https://store.steampowered.com/app/32800/Lord_of_the_Rings_War_in_the_North/)
- [Total War: SHOGUN 2](https://store.steampowered.com/app/201270/Total_War_SHOGUN_2/)
- [Saints Row: The Third](https://store.steampowered.com/app/55230/Saints_Row_The_Third/)
- [Age of Mythology: Extended Edition](https://store.steampowered.com/app/266840/Age_of_Mythology_Extended_Edition/)
- [Sniper Elite V2](https://store.steampowered.com/app/63380/Sniper_Elite_V2/)
- [Company of Heroes 2](https://store.steampowered.com/app/231430/Company_of_Heroes_2/)
- [Duke Nukem Forever](https://store.steampowered.com/app/57900/Duke_Nukem_Forever/)
- [Age of Empires II HD](https://store.steampowered.com/app/221380/Age_of_Empires_II_HD/)
- [F.E.A.R. 3](https://store.steampowered.com/app/21100/FEAR_3/)
- [Sid Meier's Civilization® V](https://store.steampowered.com/app/8930/Sid_Meiers_Civilization_V/)
- [Left 4 Dead 2](https://store.steampowered.com/app/550/Left_4_Dead_2/)
- [Lara Croft and the Guardian of Light](https://store.steampowered.com/app/35130/Lara_Croft_and_the_Guardian_of_Light/)

**Comment by kakra** [comment]
Some of those have native Linux ports... And to some of the others I have access via family library option for testing.

Lord of the Rings: War in the North is actually one I own and I'd like to be able to run.

**Comment by ahjolinna** [comment]
yes some of them have native version, but you can still install the windows version...also the demos don't seem to have native version

btw. if you don't have any games to replicate the 32bit issue that I had with dawn of war 1, its [now really cheap (-75%)](https://store.steampowered.com/sale/warhammer/) ...GOTY (main game) and Winter Assault are the only ones that have this problems other 2 expansions don't
